box2dbody.cpp

```diff
@@ -243,6 +244,34 @@ Box2DFixture *Box2DBody::at_fixture(QQmlListProperty<Box2DFixture> *list, int in
     return body->mFixtures.at(index);
 }
 
+QPointF Box2DBody::originOffsetBody()
+{
+    Q_ASSERT(mTarget);
+
+    QPointF origin = -mTarget->transformOriginPoint();
+    qreal c = qCos(-mBodyDef.angle);
+    qreal s = qSin(-mBodyDef.angle);
+
+    QPointF point = mTarget->position();
+    point.rx() += origin.x() * c - origin.y() * s - origin.x();
+    point.ry() += origin.x() * s + origin.y() * c - origin.y();
+    return point;
+}
+
+QPointF Box2DBody::originOffsetTarget()
+{
+    Q_ASSERT(mTarget);
+
+    QPointF origin = -mTarget->transformOriginPoint();
+    qreal c = qCos(-mBodyDef.angle);
+    qreal s = qSin(-mBodyDef.angle);
+
+    QPointF point = mWorld->toPixels(mBodyDef.position);
+    point.rx() -= origin.x() * c - origin.y() * s - origin.x();
+    point.ry() -= origin.x() * s + origin.y() * c - origin.y();
+    return point;
+}
```

Comment:
These functions are of course largely the same. What if you just had one function called `originOffset`, that would really just be the offset and not include the position to which it is applied, which seems to be the main difference:
```c++
QPointF Box2DBody::originOffset() const
{
    Q_ASSERT(mTarget);

    QPointF origin = -mTarget->transformOriginPoint();
    qreal c = qCos(-mBodyDef.angle);
    qreal s = qSin(-mBodyDef.angle);

    return QPointF(origin.x() * c - origin.y() * s - origin.x(),
                   origin.x() * s + origin.y() * c - origin.y());
}
```
You should be able to use this function in place of the other two with minimal extra code at the caller side.